### ISSUE TITLE
InvalidArrayOffset false positive when const array uses mixed types

### ISSUE THREAD

**Issue by ostrolucky** [vimeo/psalm#751]
```php
<?php
define('INT_CONST', 1);
define('STRING_CONST', 'string');
  
define('STUFF', [INT_CONST, STRING_CONST]);

echo STUFF[STRING_CONST];
```

> Psalm output:
> 
> ERROR: InvalidArrayOffset - 7:6 - Cannot access value on variable using offset value of 'string', expecting '0' or '1'
> 
> INFO: MixedArgument - 7:6 - Argument 1 of echo cannot be mixed, expecting string

https://getpsalm.org/r/e7430dab54

**Comment by weirdan** [comment]
I think Psalm is right in this case, since `STUFF` only has two integer keys (0 and 1). 
https://3v4l.org/S6R5V

**Comment by ostrolucky** [comment]
Indeed, this is wrong reproducer. Will try to come up with another

**Comment by ostrolucky** [comment]
This has been fixed in Psalm 1.1.4 (or 2, not really sure)
